resqpy/olio/grid_functions.py

```diff
@@ -545,45 +562,38 @@ def left_right_foursome(full_pillar_list, p_index):
     next = np.array(full_pillar_list[p_index + 1], dtype = int)
     entry = tuple(here - previous)
     exit = tuple(next - here)
-    if entry == (0, 1):
-        if exit == (-1, 0):
-            return np.array([[False, True], [True, True]], dtype = bool)
-        elif exit == (0, 1):
-            return np.array([[False, False], [True, True]], dtype = bool)
-        elif exit == (1, 0):
-            return np.array([[False, False], [True, False]], dtype = bool)
-        else:
-            raise Exception('code failure whilst taking exit sides from dubious full pillar list')
-    elif entry == (0, -1):
-        if exit == (-1, 0):
-            return np.array([[False, True], [False, False]], dtype = bool)
-        elif exit == (0, -1):
-            return np.array([[True, True], [False, False]], dtype = bool)
-        elif exit == (1, 0):
-            return np.array([[True, True], [True, False]], dtype = bool)
-        else:
-            raise Exception('code failure whilst taking exit sides from dubious full pillar list')
-    elif entry == (1, 0):
-        if exit == (0, -1):
-            return np.array([[True, False], [False, False]], dtype = bool)
-        elif exit == (1, 0):
-            return np.array([[True, False], [True, False]], dtype = bool)
-        elif exit == (0, 1):
-            return np.array([[True, False], [True, True]], dtype = bool)
-        else:
-            raise Exception('code failure whilst taking exit sides from dubious full pillar list')
-    elif entry == (-1, 0):
-        if exit == (0, -1):
-            return np.array([[True, True], [False, True]], dtype = bool)
-        elif exit == (-1, 0):
-            return np.array([[False, True], [False, True]], dtype = bool)
-        elif exit == (0, 1):
-            return np.array([[False, False], [False, True]], dtype = bool)
-        else:
-            raise Exception('code failure whilst taking exit sides from dubious full pillar list')
-    else:
+    entry_tuples_list = [(0, 1), (0, -1), (1, 0), (-1, 0)]
+    exit_tuples_list = [[(-1, 0), (0, 1), (1, 0)], [(-1, 0), (0, -1), (1, 0)], [(0, -1), (1, 0), (0, 1)],
+                        [(0, -1), (-1, 0), (0, 1)]]
+    result_arrays_list = [[
```

Comment:
Assuming this has been through yapf, you might like to disable its formatting of this list.  To do that add the following comment lines before and after the code to be spared reformatting:
```
# yapf: disable
# yapf: enable
```